Add hover and active states

kathy-t opened the pull request:
For [SEAB-2845](https://ucsc-cgl.atlassian.net/browse/SEAB-2845)
[Zeplin link](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/607da9695f1d1e9c06b6e74e)

**Navigation bar**
This picture shows the active state (user is on the Search page) and the hover state for the "About" button:
![active and hover navigation bar](https://user-images.githubusercontent.com/25287123/117197106-6891bf00-adb5-11eb-9435-2ea4fd0f8ab0.png)
Notes:
- There's no hover or active state for the Dockstore home button
- There's no active state for Docs or Forum because they open a new window. The active state is what ever page they were on previously
- Some of the buttons on the Home page, About page, or Organizations page take you to other sections of the navigation bar, so the active state is applied accordingly, depending on what page you're currently on. It does not depend on what button you click on the navigation bar. 
- Example 1: if you're on the search page (active state is Search), and you click on a workflow, the active state on Search is gone and there is no active state on the navigation bar
- Example 2: if you're on the home page, and you click the Browse Organizations button, the active state will be on the Organizations button in the navigation bar

**White links on dark background**
![hover white links on dark background](https://user-images.githubusercontent.com/25287123/117200543-92e57b80-adb9-11eb-8f7d-5c4829577432.png)

**Green accent buttons**
This shows the hover state when the mouse is hovering over the "About Dockstore" button:
![Hover for green buttons](https://user-images.githubusercontent.com/25287123/117198380-f5894800-adb6-11eb-97e3-e422aa08cb9e.png)
Other green buttons on the home page also have this hover effect, including the login button. 

I made some changes so other green mat-raised-buttons on other pages, like Organizations, also have this effect. For example, the View buttons:
![Hover state for view buttons](https://user-images.githubusercontent.com/25287123/117198872-895b1400-adb7-11eb-9823-f5b6902078e0.png)

Another example is on the Search page:
![hover search green buttons](https://user-images.githubusercontent.com/25287123/117201706-f45a1a00-adba-11eb-8bbd-39bd188e502e.png)


**Tabs**
This is how the tabs look like with no hover:
![no hover tabs](https://user-images.githubusercontent.com/25287123/117199305-08e8e300-adb8-11eb-8246-e98161562178.png)

This is how they look like with hover:
![hover tabs](https://user-images.githubusercontent.com/25287123/117199465-3afa4500-adb8-11eb-8fbc-d4d72bfc14d6.png)
I did it by changing the opacity of the tab on hover, but in my opinion it's still a little hard to tell since the font is thin. I could change the opacity to 1 on hover, but then it may look too much like an active tab. Any thoughts about this would be appreciated.

This hover effect on tabs is also applied to any other tabs on the site, like the My Workflows page.
![hover tabs my workflows](https://user-images.githubusercontent.com/25287123/117200250-4863ff00-adb9-11eb-92d8-cfa29da4f86d.png)

**Misc**

Fixed the How to Star link because I noticed the page was highlighting the word "star" which made it hard to read.

Refactored the code that highlighted the active page in this menu so it was similar to how I did it for the navigation bar:
![Active state Menu](https://user-images.githubusercontent.com/25287123/117202499-e062e800-adbb-11eb-9e62-97ed29a3e07c.png)
The appearance is still the same, but the active state is remembered even if the page is refreshed. Previously, it was lost when refreshed.

garyluu commented on src/app/about/about.component.scss:
```diff
@@ -18,10 +18,7 @@
 }
 
 button {
-  background-color: mat-color($dockstore-app-accent);
-  color: white;
-  padding: 1rem;
-  text-decoration: none;
```
👍

garyluu commented on src/app/navbar/navbar.component.ts:
```diff
@@ -69,6 +71,7 @@ export class NavbarComponent extends Logout implements OnInit {
       }
     });
     this.userQuery.extendedUserData$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((extendedUser) => (this.extendedUser = extendedUser));
+    this.navbarService.activePage$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((activePage) => (this.activePage = activePage));
```
shouldn't need to manually set and activePage. The activePage can probably be figured out using some field within activatedRoute. If activatedRoute isn't possible. this should not be a subscribe. an async pipe in the html should do

kathy-t replied:
Removed that in my latest PR. Setting active state for the navbar and menu items using `routerLinkActive` in the html now

garyluu commented on src/styles.scss:
```diff
@@ -551,6 +555,14 @@ div.button a {
 }
 
+.mat-raised-button:hover {
+  text-decoration: none;
```
it should already be none. Is there something else causing it to not be? Find and remove that instead

kathy-t replied:
My mistake, that shouldn't be there. Forgot to remove it after testing. Removed now

denis-yuen reviewed (commented):
That's a pretty thorough write-up, thanks!

> I did it by changing the opacity of the tab on hover, but in my opinion it's still a little hard to tell since the font is thin. I could change the opacity to 1 on hover, but then it may look too much like an active tab. Any thoughts about this would be appreciated.

I did do a double-take before noticing the change. That said, it might be more obvious "in motion" when the cursor is moving, I think we can run with this for now.

coverbeck reviewed (approved):
I find the distinction between active and hover in the navigation bar pretty subtle in your screenshot. I think it's partially due to the fact that the image is cropped in the PR -- in the Zepplin link the difference is more noticeable. But even there, I can't really tell which is which.

It might be obvious though when you see it in context of the mouse moving, similar to Denis' other comment.

garyluu commented on src/app/navbar/navbar.component.scss:
```diff
@@ -14,18 +14,34 @@
+}
+
+.icon:hover {
+  background-color: #112246;
```
what is this color? only used for this component and not elsewhere or in the color palette?

kathy-t replied:
It's the color for the navigation bar's hover state, according to the [zeplin](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/607da9695f1d1e9c06b6e74e) and it's only used in this component. The actual color isn't specified in the zeplin, but using the svg image given, I was able to identify the color as `#112246`. I couldn't find that color in [kim's color palette](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc699d068c46f17551f745b)

garyluu replied:
put a code comment and say whether it's darker or lighter than the ones in kim's color palette

garyluu commented on src/app/navbar/navbar.component.scss:
```diff
@@ -14,18 +14,34 @@
+}
+
+.menu-item-active {
+  background-color: #dfe0e2;
```
ditto

kathy-t replied:
This color appears to be in the gray scale of kim's color palette, but the gray scale isn't in `materialColorScheme.scss`. The color is used in 3 styles in `styles.scss` and in the about component as well

garyluu replied:
See if you can somehow specify it in the material color scheme and then reference it. if not possible via that, do something from https://sass-lang.com/documentation/variables

garyluu commented on src/app/home-page/home-logged-out/home.component.scss:
```diff
@@ -28,10 +28,7 @@
-  color: white;
-  padding: 1rem;
-  text-decoration: none;
+  padding: 0.3rem 1rem;
```
what's so special about the buttons on the logged out homepage that it has different padding than all the buttons in the entire site?

garyluu replied:
what about other pages? what padding are tehy using?

kathy-t replied:
Seems like mat-raised-buttons on other pages don't have padding (just the default padding from mat-raised-buttons)

garyluu replied:
looks like the default is 0px and 16px. if you're going to add padding for buttons, add it globally (unless there's a good reason)

kathy-t replied:
Edit: removed the additional padding for buttons on the home page and about page because it didn't make that much of a difference (pics above for comparison). It still has the default 0px and 16px padding

denis-yuen reviewed (approved):
With Gary's feedback

kathy-t commented:
Also added the grayscale from https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc699d068c46f17551f745b to the material color scheme because several of the grayscale colors are used in the UI. Removed the duplicate grayscale colors
